linux/LinuxProcess.c

```diff
@@ -381,180 +402,139 @@ void LinuxProcess_makeCommandStr(Process* this) {
+   }
 
-      mc->separateComm = true;  /* procComm a separate field */
+   if (!matchLen || (haveCommField && *cmdline)) {
+      /* cmdline will be a separate field */
+      WRITE_SEPARATOR;
    }
 
+   if (!haveCommInExe && haveCommInCmdline && !haveCommField)
+      WRITE_HIGHLIGHT(commStart, commEnd - commStart, commAttr, CMDLINE_HIGHLIGHT_FLAG_COMM);
+
    /* Display cmdline if it hasn't been consumed by procExe */
    if (*cmdline) {
       (void) stpcpyWithNewlineConversion(str, cmdline);
    }
 
-   mc->commStart = commStart;
-   mc->commEnd = commEnd;
+   #undef WRITE_SEPARATOR
+   #undef WRITE_HIGHLIGHT
 }
 
 static void LinuxProcess_writeCommand(const Process* this, int attr, int baseAttr, RichString* str) {
+   (void)baseAttr;
+
    const LinuxProcess *lp = (const LinuxProcess *)this;
    const LinuxProcessMergedCommand *mc = &lp->mergedCommand;
 
    int strStart = RichString_size(str);
 
-   int baseStart = strStart + lp->mergedCommand.baseStart;
-   int baseEnd = strStart + lp->mergedCommand.baseEnd;
-   int commStart = strStart + lp->mergedCommand.commStart;
-   int commEnd = strStart + lp->mergedCommand.commEnd;
+//   int commAttr = CRT_colors[Process_isUserlandThread(this) ? PROCESS_THREAD_COMM : PROCESS_COMM];
 
-   int commAttr = CRT_colors[Process_isUserlandThread(this) ? PROCESS_THREAD_COMM : PROCESS_COMM];
+   const bool highlightBaseName = this->settings->highlightBaseName;
+   const bool highlightSeparator = true;
+   const bool highlightDeleted = this->settings->highlightDeletedExe;
 
-   bool highlightBaseName = this->settings->highlightBaseName;
+   RichString_appendWide(str, attr, lp->mergedCommand.str);
 
-   if(lp->procExeDeleted)
-      baseAttr = CRT_colors[FAILED_READ];
+   for (size_t i = 0, hlCount = CLAMP(mc->highlightCount, 0, ARRAYSIZE(mc->highlights)); i < hlCount; i++)
+   {
+      const LinuxProcessCmdlineHighlight* hl = &mc->highlights[i];
 
-   RichString_appendWide(str, attr, lp->mergedCommand.str);
+      if (!hl->length)
+         continue;
 
-   if (lp->mergedCommand.commEnd) {
-      if (!lp->mergedCommand.separateComm && commStart == baseStart && highlightBaseName) {
-         /* If it was matched with procExe's basename, make it bold if needed */
-         if (commEnd > baseEnd) {
-            RichString_setAttrn(str, A_BOLD | baseAttr, baseStart, baseEnd - baseStart);
-            RichString_setAttrn(str, A_BOLD | commAttr, baseEnd, commEnd - baseEnd);
-         } else if (commEnd < baseEnd) {
-            RichString_setAttrn(str, A_BOLD | commAttr, commStart, commEnd - commStart);
-            RichString_setAttrn(str, A_BOLD | baseAttr, commEnd, baseEnd - commEnd);
-         } else {
-            // Actually should be highlighted commAttr, but marked baseAttr to reduce visual noise
-            RichString_setAttrn(str, A_BOLD | baseAttr, commStart, commEnd - commStart);
-         }
+      if (hl->flags & CMDLINE_HIGHLIGHT_FLAG_SEPARATOR)
+         if (!highlightSeparator)
```

Comment:
maybe combine into `if ((hl->flags & CMDLINE_HIGHLIGHT_FLAG_SEPARATOR) && !highlightSeparator)`

Reply:
intentional to illustrate the overall flag/condition structure.

Reply:
fair enough